### PR TITLE
Set FIREBASE_CONFIG env variable functions functions creates and updates

### PR DESCRIPTION
**Pull request by laurenzlong**
### Description

Set the FIREBASE_CONFIG env variable when a function is created or updated, instead of solely relying on the SDK to parse it from `.runtimeconfig.json`. This will prevent issues like https://github.com/firebase/firebase-functions/issues/669 (but only for users that update their CLI). 

I initially tried to set GCLOUD_PROJECT too but that resulted in a server error:
```
 HTTP RESPONSE BODY {"error":{"code":400,"message":"The request has errors","status":"INVALID_ARGUMENT","details":[{"@type":"type.googleapis.com/google.rpc.BadRequest","fieldViolations":[{"field":"environment_variables","description":"environment variable name GCLOUD_PROJECT is reserved by the system: it cannot be set by users"}]}]}}
```
But it's actually not a problem, since firebase-functions SDK will generate GCLOUD_PROJECT from FIREBASE_CONFIG anyhow (https://github.com/firebase/firebase-functions/blob/master/src/setup.ts#L38)

### Scenarios Tested

- Create a new function, verify in Cloud Console that FIREBASE_CONFIG env variable is set (tried Node 8 and Node 10)
- Update a function that previously was deployed with an older version of the CLI, then verify in Cloud Console that  FIREBASE_CONFIG env variable is set
- Update a function that had a custom env variable, and verified that it still persisted

### CODE REVIEW

File: src/gcp/cloudfunctions.js [hunk 1]
```diff
@@ -167,6 +169,10 @@ function _updateFunction(options) {
     data.maxInstances = Number(options.maxInstances);
     masks.push("maxInstances");
   }
+  if (options.environmentVariables) {
```
Comment (joehan) [inline src/gcp/cloudfunctions.js:172]: Same question as @mbleigh - does this overwrite any existing environment variables?

**Comment by laurenzlong** [comment]
@mbleigh @joehan Good catch! It now preserves pre-existing variables, and I verified that updates work for functions with and without preexisting custom env vars.

**Comment by samtstern** [comment]
Did something change recently in GCF that broke this?

**Comment by laurenzlong** [comment]
@samtstern Not that we know of, and we haven't been able to repro the error on our side. But this PR is good to have anyhow and will prevent future issues.

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1 @@
+- Sets FIREBASE_CONFIG environment variable during functions deploys to fix "process.env.GCLOUD_PROJECT is not set" issues during deployment or execution.
```
Comment (bkendall) [inline CHANGELOG.md:1]:
```suggestion
- Sets the `FIREBASE_CONFIG` environment variable during deploys of Cloud Functions for Firebase to fix "process.env.GCLOUD_PROJECT is not set" issues during deployment or execution.
```

**Pull request merged**